test/spec/Feature/Query/CustomMediaSpec.hs

```diff
@@ -230,6 +230,76 @@ spec = describe "custom media types" $ do
         simpleHeaders r `shouldContain` [("Content-Disposition", "attachment; filename=\"lines.csv\"")]
 
+  -- https://github.com/PostgREST/postgrest/issues/3160
+  context "using select query parameter" $ do
+    it "without select" $ do
+      request methodGet "/projects?id=in.(1,2)" (acceptHdrs "pg/outfunc") ""
+        `shouldRespondWith`
+        [str|(1,"Windows 7",1)
+            |(2,"Windows 10",1)
+            |]
+        { matchStatus  = 200
+        , matchHeaders = ["Content-Type" <:> "pg/outfunc"]
+        }
+
+    it "with fewer columns selected" $ do
+      request methodGet "/projects?id=in.(1,2)&select=id,name" (acceptHdrs "pg/outfunc") ""
+        `shouldRespondWith`
+        [str|(1,"Windows 7")
+            |(2,"Windows 10")
+            |]
+        { matchStatus  = 200
+        , matchHeaders = ["Content-Type" <:> "pg/outfunc"]
+        }
+
+    it "with columns in different order" $ do
+      request methodGet "/projects?id=in.(1,2)&select=name,id,client_id" (acceptHdrs "pg/outfunc") ""
+        `shouldRespondWith`
+        [str|("Windows 7",1,1)
+            |("Windows 10",2,1)
+            |]
+        { matchStatus  = 200
+        , matchHeaders = ["Content-Type" <:> "pg/outfunc"]
+        }
+
+    it "with computed columns" $ do
+      request methodGet "/items?id=in.(1,2)&select=id,always_true" (acceptHdrs "pg/outfunc") ""
+        `shouldRespondWith`
+        [str|(1,t)
+            |(2,t)
+            |]
+        { matchStatus  = 200
+        , matchHeaders = ["Content-Type" <:> "pg/outfunc"]
+        }
+
+    -- TODO: Embeddings should not return JSON. Arrays of record would be much better.
+    it "with embedding" $ do
+      request methodGet "/projects?id=in.(1,2)&select=*,clients(id)" (acceptHdrs "pg/outfunc") ""
+        `shouldRespondWith`
+        [str|(1,"Windows 7",1,"{""id"": 1}")
+            |(2,"Windows 10",1,"{""id"": 1}")
+            |]
+        { matchStatus  = 200
+        , matchHeaders = ["Content-Type" <:> "pg/outfunc"]
+        }
+
+    it "will fail for specific aggregate with fewer columns" $ do
+      request methodGet "/lines?select=id" (acceptHdrs "application/vnd.twkb") ""
+        `shouldRespondWith` 406
+
+    it "will fail for specific aggregate with more columns" $ do
+      request methodGet "/lines?select=id,name,geom,id" (acceptHdrs "application/vnd.twkb") ""
+        `shouldRespondWith` 406
+
+    it "will fail for specific aggregate with columns in different order" $ do
+      request methodGet "/lines?select=name,id,geom" (acceptHdrs "application/vnd.twkb") ""
+        `shouldRespondWith` 406
+
+    -- This is just because it would be hard to detect this case, so we better error in this case, too.
+    it "will fail for specific aggregate with columns in same order" $ do
+      request methodGet "/lines?select=id,name,geom" (acceptHdrs "application/vnd.twkb") ""
+        `shouldRespondWith` 406
```

Comment:
Yes, these look good :+1:

Reply:
@wolfgangwalther LGTM. The only purpose was to allow having a generic and particular handler for a media type. If the function name has to change for that it's fine.

> Would you agree to just remove those tests (or change the expected outcome to the error message),

How about just renaming the functions and not removing the tests?

Reply:
> How about just renaming the functions and not removing the tests?

Just tried that and it does work.